File: Data Structures & Algorithms/longest-substring-without-duplicates/submission-0.js
class Solution {
    /**
     * @param {string} s
     * @return {number}
     */
    lengthOfLongestSubstring(s) {
        let l = 0;
        let ans = 0;
        let win = new Set();

        for (let r = 0; r < s.length; r++) {
            while (win.has(s.charAt(r))) {
                win.delete(s.charAt(l));
                l++;
            }
            win.add(s.charAt(r));
            ans = Math.max(ans, r - l + 1);
        }

        return ans;
    }
}

File: Data Structures & Algorithms/buy-and-sell-crypto/submission-0.java
class Solution {
    public int maxProfit(int[] prices) {
        int max = 0;
        int L = 0;

        for (int R = L + 1; R < prices.length; R++) {
            if (prices[L] >= prices[R]) {
                L = R;
            }

            max = Math.max(max, prices[R] - prices[L]);
        }

        return max;
    }
}
